src/Data/Array/Accelerate/Error.hs

```diff
@@ -33,13 +33,13 @@ data Check = Bounds | Unsafe | Internal
 -- | Issue an internal error message
 --
 internalError :: HasCallStack => String -> a
-internalError = withFrozenCallStack $ error Internal
+internalError = withFrozenCallStack `id` error Internal
```

Comment:
Doesn't compile:

```
.../src/Data/Array/Accelerate/Error.hs:36:17: error:
    • Couldn't match type ‘?callStack::CallStack’ with ‘[Char]’
      Expected type: (String -> String -> a) -> String -> a
        Actual type: (HasCallStack => String -> a) -> String -> a
    • In the first argument of ‘id’, namely ‘withFrozenCallStack’
      In the expression: withFrozenCallStack `id` error Internal
      In an equation for ‘internalError’:
          internalError = withFrozenCallStack `id` error Internal
   |
36 | internalError = withFrozenCallStack `id` error Internal
   | 
```

Reply:
Oh is that because of the special handling of `($)` in GHC?

Reply:
Yes, `($)` is levity-polymorphic. I'm working on a fix.